### PR TITLE
Refactor CPU_EXTENSIONS logic for MSVC with Clang

### PR DESCRIPTION
**Pull request by waahm7**
*Description of changes:*
`visualc_crc32c_sse42.c` fails to compile on MSVC with Clang because we don't compile it with the AVX or AVX2 flag. Since inline assembly is supported by Clang even on MSVC, first check if we can use that. This will fix the https://github.com/awslabs/aws-checksums/pull/66.

TODO: fix warnings and add CI for MSVC with Clang.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -58,17 +58,16 @@ file(GLOB AWS_ARCH_SRC
         )
 
 if (USE_CPU_EXTENSIONS)
-    if (MSVC AND AWS_ARCH_INTEL)
+    # First, check if inline assembly is available. Inline assembly can also be supported by MSVC if the compiler in use is Clang.
+    if(AWS_ARCH_INTEL AND AWS_HAVE_GCC_INLINE_ASM)
+        file(GLOB AWS_ARCH_SRC
+                "source/intel/asm/*.c"
+            )
+    elseif (MSVC AND AWS_ARCH_INTEL)
```
Comment (graebm) [inline CMakeLists.txt:66]: nitpick: I realize the code was already like this, but I find this logic hard to follow:
```
if INTEL AND ASM:
    ...
else if MSVC AND INTEL:
    ...
```

can we change this to:
```
if INTEL:
    if ASM:
        ...
    else if MSVC:
        ...
```

Comment (waahm7) [reply CMakeLists.txt]: Thanks, updated.

**Review by graebm — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -58,17 +58,18 @@ file(GLOB AWS_ARCH_SRC
...
-                "source/intel/visualc/*.c"
-        )
-
-        source_group("Source Files\\intel\\visualc" FILES ${AWS_ARCH_SRC})
-
-    elseif(AWS_ARCH_INTEL AND AWS_HAVE_GCC_INLINE_ASM)
-        file(GLOB AWS_ARCH_SRC
-                "source/intel/asm/*.c"
+    if(AWS_ARCH_INTEL)
+        # First, check if inline assembly is available. Inline assembly can also be supported by MSVC if the compiler in use is Clang.
```
Comment (graebm) [inline CMakeLists.txt:62]: dogs and cat
living together
mass hysteria
